lib/std/wrapnils.nim

```diff
@@ -15,6 +15,20 @@ doAssert ?.default(seq[int])[3] == default(int)
 
 import macros
 
+proc checkNil*[T: ptr|ref|cstring|proc](a: T): T {.inline.} =
```

Comment:
Could you elaborate on what this does a bit more? I thought the whole point of `?.` was to return a default value in case of a nil expression.